lib/preface_specs/freer_selective.mli

```diff
@@ -0,0 +1,110 @@
+(** A [Freer selective] allows you to build a {e rigid}
+    {!module:Preface_specs.Selective} from a given arbitrary type. *)
+
+(** Such {!module:Preface_specs.selective} is equiped with and additional
+    function for [promoting] values from the underlying type into the
+    [Freer selective] and a [Natural transformations] for transforming the value
+    of the [Freer selective] to an other {!module:Preface_specs.Selective} or to
+    a {!module:Preface_specs.Monoid}. *)
+
+(** {2 Note about complexity}
```

Comment:
It would be nice to be more specific here: what's the complexity of functions that this construction provides?

Reply:
Fixed on the last commit, adding : `[map] and [select] (so it also applies to derived combinators) have [O(n)]`, thanks a lot for your feedback. After working on Decidable and Distributive, I'll take a look on cheaper implementation based on articles that you join in the documentation (faster Free Applicative).

Reply:
Cool, sounds great!